userspace/plugin/plugin_api.h

```diff
@@ -1053,6 +1053,25 @@ typedef struct {
 		ss_plugin_rc (*set_async_event_handler)(ss_plugin_t* s,
 		                                        ss_plugin_owner_t* owner,
 		                                        const ss_plugin_async_event_handler_t handler);
+
```

Comment:
I don't fully enjoy the fact that this is tied to the async events capability. For instances, certain plugins may collect some state through `parse_event` and make it available through the table API too, but maybe will never emit async events.

However, I see the point. It also makes sense to "recycle" `get_async_event_sources` and `get_async_events`, since they are indeed useful for this use case. Given that this symbol is optional, my interpretation is that defining `set_async_event_handler` will be a requirement even if not used. If so, we'll have the opportunity later on to change our minds and loosen the requirements without incurring in a breaking change. So I think we can sleep peacefully with this for now 👍